pymc3/stats.py

```diff
@@ -509,6 +509,9 @@ def df_summary(trace, varnames=None, stat_funcs=None, extend=False, include_tran
         else:
             varnames = [name for name in trace.varnames if not name.endswith('_')]
 
+    if batches is None:
+        batches = min([100, len(trace)])
```

Comment:
We could do that, but then batches could potentially take on a different value for each variable, which seems problematic (and would also require defining the stat func lambdas inside the varnames loop, since the values are bound by then). I don't really see a problem with doing it this way, since this is really just designed to catch the corner case where someone is testing or debugging a model with a tiny number of samples. It's a safe approach, in the sense that len(trace) is guaranteed to give a smaller value than the number of flattened samples. So I'd prefer to keep it this way...